lib/WorseReflection/Tests/Inference/generics/method_returns_templated_generic.test

```diff
@@ -0,0 +1,37 @@
+<?php
+
+namespace Bar;
+
+/**
+ * @template T
+ * @extends IteratorAggregate<T>
+ */
+interface Collection
+{
+    /**
+     * @return T
+     */
+    public function get();
+}
+
+/**
+ * @template T
+ */
+class Test
+{
+    /**
+     * @param T $foo
+     */
+    public function __construct($foo)
+    {
+    }
+    /**
+     * @return Collection<T>
+     */
+    public function foobar(): Collection
+    {
+    }
+}
+
+$foo = new Test(new \stdClass());
+wrAssertType('stdClass', $foo->foobar()->get());
```

Comment:
this test worked as expected first time, keeping though as the test that covers this isn't so obvious.